File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/dtos/PaymentDto.java
package com.example.dalathasfarm.dtos;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PaymentDto {
    @NotNull(message = "Amount is required")
    @Positive(message = "Amount must be positive")
    private long amount;

    private String bankCode;

    @NotNull(message = "Language is required")
    private String language;
}

File: android/lib/dtos/payment_dto.dart
class PaymentDto{
  final double amount;
  final String language;
  
  PaymentDto({
    required this.amount,
    required this.language,
  });

  Map<String, dynamic> toJson() {
    return {
      'amount': amount,
      'language': language,
    };
  }
}